app/src/main/java/dev/davidv/translator/BackgroundTranslationService.kt

```diff
@@ -0,0 +1,85 @@
+        val translationService = TranslationService(settingsManager, filePathManager)
+        val languageDetector = LanguageDetector()
+        val imageProcessor = ImageProcessor(this, OCRService(filePathManager))
+        translationCoordinator = TranslationCoordinator(this, translationService, languageDetector, imageProcessor, settingsManager, false)
+    }
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        if (intent?.action == "dev.davidv.translator.action.TRANSLATE_TEXT") {
+            val textToTranslate = intent.getStringExtra("text_to_translate")
+            val fromLanguageStr = intent.getStringExtra("from_language")
+            val toLanguageStr = intent.getStringExtra("to_language")
+            val fromLanguage = fromLanguageStr?.let { lng -> Language.entries.find { it.code == lng } }
+            val toLanguage = toLanguageStr?.let { lng -> Language.entries.find { it.code == lng } }
+            val receiver = intent.getParcelableExtra<Messenger>("result_receiver")
+            Log.d(TAG, "translate txt: $textToTranslate, $fromLanguage, $toLanguage, cb = ${receiver != null}")
+
+            if (textToTranslate != null && receiver != null) {
+                CoroutineScope(Dispatchers.IO).launch {
+                    val from = fromLanguage ?: translationCoordinator.detectLanguageRobust(textToTranslate, null)
+                    if (from != null) {
+                        val to = toLanguage ?: SettingsManager(applicationContext).settings.value.defaultTargetLanguage
+                        val result = translationCoordinator.translateText(from, to, textToTranslate)
+                        val translatedText = when (result) {
+                            is TranslationResult.Success -> result.result.translated
+                            is TranslationResult.Error -> "Error: " + result.message
+                            null -> "Error: Translation failed"
+                        }
+                        val bundleKey = when (result) {
+                          is TranslationResult.Success -> "translated_text"
+                          else -> "error"
+                        }
+                        Log.d(TAG, "$bundleKey: $translatedText")
+                        val bundle = Bundle()
+                        bundle.putString(bundleKey, translatedText)
+                        val msg = Message.obtain()
+                        msg.data = bundle
+                      receiver.send(msg)
+                    } else {
+                        val bundle = Bundle()
+                        bundle.putString("translated_text", "Error: Could not detect language")
+                        Log.d(TAG, "Error: Could not detect language")
+                      val msg = Message.obtain()
+                      msg.data = bundle
+                      receiver.send(msg)
+                    }
+                }
```

Comment:
I think a `stopSelf()` is missing here, no? otherwise with START_NOT_STICKY it's relying on system GC to get rid of the service

Reply:
Added.